### PR TITLE
MLPAB-88 - IDE setup

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Make Goland run app and tests in docker

Fixes MLPAB-88

## Approach

- Add run configuration for app
- Add run configuration for cypress testing
- Refactor Dockerfile with multistage targets for development and production

## Learning

Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Modernising LPA service

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* ~I have added welsh translation tags and updated translation files~
* ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by acsauk — APPROVED** [review]
:shipit:

File: Makefile [hunk 1]
```diff
@@ -1,2 +1,11 @@
 go-test:
 	find . -name go.mod -execdir go test ./... -race -covermode=atomic -coverprofile=coverage.out \;
+
+run-cypress-dc:
+	docker compose -f /Users/andrewpearce/workspace/opg-modernising-lpa/docker-compose.yml \
+	-f /Users/andrewpearce/workspace/opg-modernising-lpa/docker-compose.override.yml \
+	-f /Users/andrewpearce/workspace/opg-modernising-lpa/docker-compose.testing.yml \
```
Comment (acsauk) [inline Makefile:7]: Ah missed this before approving - these will need to be changed to `./`

**Review by acsauk — CHANGES_REQUESTED** [review]
Filepaths to update then ready to merge 👍

File: Dockerfile [hunk 2]
```diff
@@ -24,11 +24,23 @@ COPY /app .
...
-FROM alpine:3.16.1
+FROM build-env as development
+
+WORKDIR /go/bin
+COPY --from=build-env /app .
+COPY --from=asset-env /app/web/static web/static
+COPY web/template web/template
+
+# Live reload for Go
+RUN go install github.com/cosmtrek/air@latest
```
Comment (hawx) [inline Dockerfile:35]: if you moved this above the `COPY`s it would be able to cache the layer (I think)

File: Makefile [hunk 3]
```diff
@@ -1,2 +1,10 @@
 go-test:
 	find . -name go.mod -execdir go test ./... -race -covermode=atomic -coverprofile=coverage.out \;
+
+run-cypress-dc:
+	docker compose -f ./opg-modernising-lpa/docker-compose.yml \
```
Comment (hawx) [inline Makefile:5]: should these paths be prefixed with `./opg-modernising-lpa/`?

**Review by hawx — COMMENTED** [review]
I don't use Goland so personally would prefer not to include IDE config but :shrug: it's only another hidden directory

File: Dockerfile (hunk 2 above)
Comment (andrewpearce-digital) [reply Dockerfile]: thanks, moved.

File: Makefile (hunk 3 above)
Comment (andrewpearce-digital) [reply Makefile]: yep, third time lucky 😄

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**